tests/conftest.py

```diff
@@ -541,3 +544,120 @@ def _factory(process: Process, state: plumpy.ProcessState = plumpy.ProcessState.
         return node
 
     return _factory
+
+
+@wrapt.decorator
+def suppress_deprecations(wrapped, _, args, kwargs):
+    """Decorator that suppresses all ``DeprecationWarning``s."""
+    with warnings.catch_warnings():
+        warnings.filterwarnings('ignore', category=DeprecationWarning)
+        return wrapped(*args, **kwargs)
+
+
+class EntryPointManager:
+    """Manager to temporarily add or remove entry points."""
+
+    @staticmethod
+    def eps():
+        return module_entry_point.eps()
+
+    @staticmethod
+    def _validate_entry_point(entry_point_string: str | None, group: str | None, name: str | None) -> tuple[str, str]:
+        """Validate the definition of the entry point.
+
+        :param entry_point_string: Fully qualified entry point string.
+        :param name: Entry point name.
+        :param group: Entry point group.
+        :returns: The entry point group and name.
+        :raises TypeError: If `entry_point_string`, `group` or `name` are not a string, when defined.
+        :raises ValueError: If `entry_point_string` is not defined, nor a `group` and `name`.
+        :raises ValueError: If `entry_point_string` is not a complete entry point string with group and name.
+        """
+        if entry_point_string is not None:
+            try:
+                group, name = module_entry_point.parse_entry_point_string(entry_point_string)
+            except TypeError as exception:
+                raise TypeError('`entry_point_string` should be a string when defined.') from exception
+            except ValueError as exception:
+                raise ValueError('invalid `entry_point_string` format, should `group:name`.') from exception
+
+        if name is None or group is None:
+            raise ValueError('neither `entry_point_string` is defined, nor `name` and `group`.')
+
+        type_check(group, str)
+        type_check(name, str)
+
+        return group, name
+
+    @suppress_deprecations
+    def add(
+        self,
+        value: type | str,
+        entry_point_string: str | None = None,
+        *,
```

Comment:
The `*` operator signifies that all arguments after that are *keyword-only* arguments, i.e., they can only be specified using keywords. See this [PEP](https://peps.python.org/pep-3102/). This is useful for this interface where `group` and `name` are mutually exclusive with `entry_point_string`. With this syntax a user calls either
```python
entry_points.add('aiida.parsers:core.base')
```
or
```python
entry_points.add(group='aiida.parsers', name='core.base')
```

Reply:
I see, thanks! Then maybe also use this for `_validate_entry_point` for the exclusive reason I mentioned above?

Reply:
The `_validate_entry_point` is just an internal private method to perform the validation. That is also the reason I didn't use the `*` method, since here it is not really necessary since only the class itself is calling this. No real point then to add it here.

Reply:
That makes sense. 👍🏻